pkg/controller/worker/machines.go

```diff
@@ -204,6 +205,19 @@ func (w *workerDelegate) generateMachineConfig() error {
 			machineClassSpec["labels"] = map[string]string{corev1.LabelZoneFailureDomain: zone}
 			machineClassSpec["secret"].(map[string]interface{})["labels"] = map[string]string{v1beta1constants.GardenerPurpose: genericworkeractuator.GardenPurposeMachineClass}
 
+			// if the shoot K8s version is greater than or equal to 1.22 then
+			// enable source destination check on AWS instances by default
+			shootK8sVersion := w.cluster.Shoot.Spec.Kubernetes.Version
+
+			k8sVersionGreaterOrEqualThan122, err := versionutils.CompareVersions(shootK8sVersion, ">=", "1.22")
+			if err != nil {
+				return err
+			}
+
+			if k8sVersionGreaterOrEqualThan122 {
+				machineClassSpec["srcAndDstChecksEnabled"] = false
```

Comment:
```golang
			// if the shoot K8s version is greater than or equal to 1.22 then
			// enable source destination check on AWS instances by default
```

Shouldn't it be ?
```suggestion
				machineClassSpec["srcAndDstChecksEnabled"] = true
```

Reply:
They should be disabled by default, they already are enabled by default.

Reply:
Ah, my bad! The code is right. The comment is wrong. I will update it.